BUG: Series.between is too strict

**Describe the bug**
`pandas.Series.between` doesn't allow `Series` as input.

**To Reproduce**

```
import pandas as pd

r = pd.Series([1, 2, 3])
s = pd.Series([0, 1, 2])
t = pd.Series([2, 3, 4])
between = r.between(s, t)
```

When executing `mypy test.py`, the following error message gets thrown:
```
❯ mypy test.py
test.py:6: error: Argument 1 to "between" of "Series" has incompatible type "Series[Any]"; expected "Union[str, bytes, date, datetime, timedelta, bool, int, float, Timestamp, Timedelta, complex, Sequence[Any]]"  [arg-type]
test.py:6: error: Argument 2 to "between" of "Series" has incompatible type "Series[Any]"; expected "Union[str, bytes, date, datetime, timedelta, bool, int, float, Timestamp, Timedelta, complex, Sequence[Any]]"  [arg-type]
Found 2 errors in 1 file (checked 1 source file)
```


**Please complete the following information:**
 - OS: macOS
 - OS Version: Ventura
 - python version: 3.11
 - version of type checker: `mypy 1.0.0`
 - version of installed `pandas-stubs`: `pandas-stubs 1.5.3.230203`

--- a/pandas-stubs/core/series.pyi
+++ b/pandas-stubs/core/series.pyi
@@ -97,6 +97,7 @@ from pandas._typing import (
     JsonSeriesOrient,
     Level,
     ListLike,
+    ListLikeU,
     MaskType,
     NaPosition,
     QuantileInterpolation,
@@ -935,8 +936,8 @@ class Series(IndexOpsMixin, NDFrame, Generic[S1]):
     def isin(self, values: Iterable | Series[S1] | dict) -> Series[_bool]: ...
     def between(
         self,
-        left: Scalar | Sequence | Series,
-        right: Scalar | Sequence | Series,
+        left: Scalar | ListLikeU,
+        right: Scalar | ListLikeU,
         inclusive: Literal["both", "neither", "left", "right"] = ...,
     ) -> Series[_bool]: ...
     def isna(self) -> Series[_bool]: ...

--- a/pandas-stubs/core/series.pyi
+++ b/pandas-stubs/core/series.pyi
@@ -935,8 +935,8 @@ class Series(IndexOpsMixin, NDFrame, Generic[S1]):
     def isin(self, values: Iterable | Series[S1] | dict) -> Series[_bool]: ...
     def between(
         self,
-        left: Scalar | Sequence,
-        right: Scalar | Sequence,
+        left: Scalar | Sequence | Series,
+        right: Scalar | Sequence | Series,
         inclusive: Literal["both", "neither", "left", "right"] = ...,
     ) -> Series[_bool]: ...
     def isna(self) -> Series[_bool]: ...

--- a/tests/test_series.py
+++ b/tests/test_series.py
@@ -622,6 +622,15 @@ def test_types_compare() -> None:
     s2.compare(s1, align_axis="columns", keep_shape=True, keep_equal=True)
 
 
+def test_types_between() -> None:
+    s1 = pd.Series([1, 2, 3])
+    s2 = pd.Series([0, 1, 2])
+    s3 = pd.Series([2, 3, 4])
+    s1.between(0, 2)
+    s1.between([0, 1, 2], [2, 3, 4])
+    s1.between(s2, s3)
+
+
 def test_types_agg() -> None:
     s = pd.Series([1, 2, 3], index=["col1", "col2", "col3"])
     check(assert_type(s.agg("min"), Any), np.int64)

--- a/tests/test_series.py
+++ b/tests/test_series.py
@@ -626,9 +626,9 @@ def test_types_between() -> None:
     s1 = pd.Series([1, 2, 3])
     s2 = pd.Series([0, 1, 2])
     s3 = pd.Series([2, 3, 4])
-    s1.between(0, 2)
-    s1.between([0, 1, 2], [2, 3, 4])
-    s1.between(s2, s3)
+    check(assert_type(s1.between(0, 2), "pd.Series[bool]"), pd.Series, bool)
+    check(assert_type(s1.between([0, 1, 2], [2, 3, 4]), "pd.Series[bool]"), pd.Series, bool)
+    check(assert_type(s1.between(s2, s3), "pd.Series[bool]"), pd.Series, bool)
 
 
 def test_types_agg() -> None:

--- a/tests/test_series.py
+++ b/tests/test_series.py
@@ -627,7 +627,11 @@ def test_types_between() -> None:
     s2 = pd.Series([0, 1, 2])
     s3 = pd.Series([2, 3, 4])
     check(assert_type(s1.between(0, 2), "pd.Series[bool]"), pd.Series, bool)
-    check(assert_type(s1.between([0, 1, 2], [2, 3, 4]), "pd.Series[bool]"), pd.Series, bool)
+    check(
+        assert_type(s1.between([0, 1, 2], [2, 3, 4]), "pd.Series[bool]"),
+        pd.Series,
+        bool,
+    )
     check(assert_type(s1.between(s2, s3), "pd.Series[bool]"), pd.Series, bool)
 
 

